pages/index/IndexHandler.inc.php

```diff
@@ -113,8 +112,22 @@ function _displayPressIndexPage($press, &$templateMgr) {
 		}
 
 		// Display Featured Books
-		$displayFeaturedBooks = $press->getSetting('displayFeaturedBooks');
-		$templateMgr->assign('displayFeaturedBooks', $displayFeaturedBooks);
+		if ($press->getSetting('displayFeaturedBooks')) {
+			$featureDao = DAORegistry::getDAO('FeatureDAO');
+			$featuredMonographIds = $featureDao->getSequencesByAssoc(ASSOC_TYPE_PRESS, $press->getId());
+			if (!empty( $featuredMonographIds)) {
+				$featuredMonographs = array();
+				$publishedMonographDao = DAORegistry::getDAO('PublishedMonographDAO');
+				$publishedMonographs = $publishedMonographDao->getByPressId($press->getId());
+				$publishedMonographs = $publishedMonographs->toAssociativeArray();
+				foreach($featuredMonographIds as $key => $val) {
```

Comment:
Better, when possible, to use iterators rather than converting to arrays. In this case, it would be something like...
````
while ($publishedMonographs = $publishedMonographs->next()) {
    ...
}
````
This is less wasteful on memory, since converting to an array means all objects need to exist in memory at once.